packages/core/src/module-resolver.ts

```diff
@@ -306,7 +306,7 @@ export class Resolver {
 
     let pkg = this.owningPackage(match.filename);
     if (pkg) {
-      let rel = withoutJSExt(explicitRelative(pkg.root, match.filename));
+      let rel = explicitRelative(pkg.root, match.filename);
```

Comment:
To drop `withoutJSExt` here you must replace it with the search over all resolvable extensions. That's what's causing the failure in the `engine-relative resolving > file exists in both app-js and fastboot-js` test.

I would suggest taking your implementation in `searchAppTree` and making it a utility method that you can use here too.

Reply:
Nice one, I completely missed that there was something else checking the mergemap 👍